File: parentconnect/lib/models/utilisateur.dart
class Utilisateur {

  final int id;

  final String nomUtilisateur;
  final String prenomUtilisateur;
  final String emailUtilisateur;
  final String roleUtilisateur;

  Utilisateur({
    required this.id,
    required this.nomUtilisateur,
    required this.prenomUtilisateur,
    required this.emailUtilisateur,
    required this.roleUtilisateur,
  });

  factory Utilisateur.fromJson(
    Map<String, dynamic> json,
  ) {
    return Utilisateur(
      id: json["id"],

      nomUtilisateur:
          json["nomUtilisateur"] ?? "",

      prenomUtilisateur:
          json["prenomUtilisateur"] ?? "",

      emailUtilisateur:
          json["emailUtilisateur"] ?? "",

      roleUtilisateur:
          json["roleUtilisateur"] ?? "",
    );
  }
}

File: ServerParent/src/models/utilisateur.js
module.exports = (sequelize, DataTypes) => {
  const Utilisateurs = sequelize.define("Utilisateurs", {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    nomUtilisateur: {
      type: DataTypes.STRING,
      allowNull: false
    },
    prenomUtilisateur: {
      type: DataTypes.STRING,
      allowNull: false
    },
    emailUtilisateur: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true
    },
    motDePasse: {
      type: DataTypes.STRING,
      allowNull: false
    },
    roleUtilisateur: {
      type: DataTypes.STRING, // Aligné sur la migration
      allowNull: false
    }
  }, {
    tableName: 'Utilisateurs' // Force Sequelize à utiliser exactement le nom de ta migration
  });

  return Utilisateurs;
};